test/Interop/Cxx/stdlib/overlay/custom-sequence.swift

```diff
@@ -0,0 +1,39 @@
+// RUN: %target-run-simple-swift(-I %S/Inputs -Xfrontend -enable-experimental-cxx-interop)
+//
+// REQUIRES: executable_test
+// REQUIRES: OS=macosx || OS=linux-gnu
+
+import StdlibUnittest
+import CustomSequence
+import std
+
+var CxxSequenceTestSuite = TestSuite("CxxSequence")
+
+extension SimpleSequence.ConstIterator: UnsafeCxxInputIterator {
+  // This should probably be synthesized from operator++.
+  public func successor() -> Self {
+    return SimpleSequence.ConstIterator(pointee + 1)
+  }
+
+  // This shouldn't be required, since operator== is defined on the C++ side.
+  // However, without this decl it fails to typecheck, likely due to a lookup issue.
+  public static func ==(lhs: SimpleSequence.ConstIterator, rhs: SimpleSequence.ConstIterator) -> Bool {
+    return lhs.pointee == rhs.pointee
```

Comment:
Yeah, ideally we shouldn't need this at all: we import `operator==` for this type and it is shown in the module interface correctly, but lookup can't find it. I'm comparing the pointees here since it doesn't really matter for a test. I'll remove this func entirely once we fix the lookup issue.